index.js

```diff
@@ -43,12 +154,20 @@ module.exports = {
 		}
 
 		for (let i = 0; i < pathArray.length; i++) {
-			object = object[pathArray[i]];
+			const key = pathArray[i];
+			const index = Number.parseInt(key, 10);
+
+			// Disallow string indexes
+			if (!Number.isInteger(key) && Array.isArray(object) && !Number.isNaN(index) && object[index] === object[key]) {
+				object = i === pathArray.length - 1 ? undefined : null;
```

Comment:
`null` is used since the rest of the code already recognises what it means

Reply:
Maybe add a code comment. It might be tempting to try to remove the `null` sometime in the future.